src/System.Private.CoreLib/src/System/Threading/Thread.CoreRT.Unix.cs

```diff
@@ -53,6 +53,7 @@ private static void OnThreadExit()
                 int state = currentThread._threadState;
                 if ((state & (int)(ThreadState.Stopped | ThreadState.Aborted)) == 0)
                 {
+                    // ThreadStore.DecrementRunningForeground(currentThread);
```

Comment:
I'm not sure whether we need it here or not. Logically we could have this callback invoked for some thread which was unable to shutdown on its own hence no `ThreadStore.DecrementRunningForeground(currentThread);` [in finally block](https://github.com/dotnet/corert/pull/8364/files#diff-804e272d32ad37bc2d1b69306f04906eR464) But when I leave this uncommented I'm getting a bugfest on top200 tests.